terraform/Environments/dpc/data.tf

```diff
@@ -0,0 +1,5 @@
+data "archive_file" "lambda_zip" {
+  type        = "zip"
+  output_path = local.lambda_zip
+  source_file = local.lambda_src
+}
```

Comment:
As noted in the ticket, the application teams should be able to push lambda code independently. To do so, let's configure the lambda to pull the zip from an S3 bucket. We can create the S3 bucket in this terraform, but leave the uploading of the zip to that bucket out of this code.